app/scripts/views/components/barChart/barChart.js

```diff
@@ -21,6 +21,7 @@
         var val = d[data_.attrId];
         var _min;
         var _max;
+        var endNumIndex = opts_.xDomain.length - 1; // when data doesn't have NA
```

Comment:
If xDomain is empty array, it means no ticks show on the chart. No matter how users select, it always shows "Invalid Selection".